aspect ratio updating

gsbernstein opened the pull request:
- new function removeConstraintWithIdentifier
 - new optional parameter on `aspectRatio` func `replacingExisting`

gsbernstein committed:
aspect ratio updating

ericgroom commented on constrain/Classes/AspectRatio.swift:
```diff
@@ -12,14 +12,16 @@ public extension Constraints {
+    /// Multiplier is not editable, so you need to replace this constraint to change the ratio, hence the optional parameter
     @discardableResult
-    func aspectRatio(_ ratio: CGFloat, by relationship: Relationship = .equal, priority: UILayoutPriority = .required) -> Constraints {
+    func aspectRatio(_ ratio: CGFloat, by relationship: Relationship = .equal, priority: UILayoutPriority = .required, replacingExisting: Bool = false) -> Constraints {
```
Do you think instead of adding this parameter that it could be the default behavior?

gsbernstein replied:
It could work, but there are potentially legitimate cases where you could have more than one aspect ratio simultaneously, like different priority and an inequality. And it saves the extra step of removing a nonexistant one in the most common use case where you're not changing it.

gsbernstein committed:
aspect ratio updating (#1)

new function removeConstraintWithIdentifier
new optional parameter on aspectRatio func replacingExisting, defaults to true
removeConstraintsWithIdentifier searches all existing constraints, not just this instance of the Constraints helper
allParticipatingConstraints extension to aid with this
Don't add viewName to the constraint identifier unless it's been manually set (more deterministic)

# Conflicts:
#	constrain/Classes/AspectRatio.swift
#	constrain/Classes/Modification.swift

gsbernstein commented:
@anconaesselmann added some new changes from my fork, lmk what you think
